code-scanning/scorecard.yml

```diff
@@ -59,14 +59,15 @@ jobs:
       # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
       # format to the repository Actions tab.
       - name: "Upload artifact"
-        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
```

Comment:
```suggestion
        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
```

Reply:
On the Scorecard side, v3.1.3 is fine by me. 

Are there any expected issues on GitHub's end from `v3.1.3` specifying node16? Given the [Spring 2024 transition](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/)?

I see there's also a [v3-node20 tag](https://github.com/actions/upload-artifact/releases/tag/v3-node20), but due to some Scorecard imposed workflow restrictions when `publish_results: true` I would need to make changes to support that tag since it doesn't live in `main`.

Reply:
@spencerschrock That's a great callout and a note to my future self that we may need an additional tag for semantic versioning.

I am curious about the consideration you mentioned in your backend. Would this not need to be changed regardless as we may always introduce a new tag to fix bugs in prior versions? For instance, a 3.1.9 could be created in the future to address some newly discovered issue, which would be a divergence from `main`.

Reply:
When `publish_results` is `true`, we want to check the workflow is generating legitimate scores before accepting them, as GitHub Actions allows the pinned SHA to be from the repo or a fork. So we don't want a forked `ossf/scorecard-action` (or similar) to generate bogus scores.

This is specifically something we check only when pinned to a hash, so someone could always pin to a tag (e.g. `@v3.1.9`).

Several of us have looked at the available APIs and there's no perfect option for "does this commit belong to the repo or a fork?". https://github.com/orgs/community/discussions/78161

Reply:
I've made the proposed changes to this workflow. 
I still need to merge https://github.com/ossf/scorecard-webapp/pull/599 on my side.

Reply:
Merged CodeQL into action/workflow from SHA f4dbb8424c
Branch: main
965022-oss-patch3